integration-tests/pkg/types/network.go

```diff
@@ -1,103 +1,169 @@
+
+	if conn.CloseTimestamp == nil {
+		conn.CloseTimestamp = nilTimestamp
 	}
 
-	ipNetwork := utils.IPNetworkFromCIDRBytes(ipNetworkData)
-	prefixLen := ipNetwork.PrefixLen()
-	// If this is IPv4 and the prefix length is 32 or this is IPv6 and the prefix length
-	// is 128 this is a regular IP address and not a CIDR block
-	if (ipNetwork.Family() == utils.IPv4 && prefixLen == byte(32)) ||
-		(ipNetwork.Family() == utils.IPv6 && prefixLen == byte(128)) {
-		peerId.Address = ipNetwork.IP()
-	} else {
-		peerId.IPNetwork = ipNetwork
+	if conn.CloseTimestamp != nil {
+		conn.CloseTimestamp = notNilTimestamp
 	}
-	return peerId.String()
 }
 
-func (n *NetworkInfo) String() string {
-	return fmt.Sprintf("%s|%s|%s|%s|%s",
-		n.LocalAddress,
-		n.RemoteAddress,
-		n.Role,
-		n.SocketFamily,
-		n.CloseTimestamp)
+// EqualVT is not called directly because it returns false in cases that we don't want it to, for example
+// when both CloseTimestamp are nil, or when they have different non-nil values.
+func EqualNetworkConnection(conn1 sensorAPI.NetworkConnection, conn2 sensorAPI.NetworkConnection) bool {
+	adjustNetworkConnectionForComparison(&conn1)
+	adjustNetworkConnectionForComparison(&conn2)
+
+	return conn1.EqualVT(&conn2)
+
 }
 
-func (n *NetworkInfo) IsActive() bool {
-	// no close timestamp means the connection is open, and active
-	return n.CloseTimestamp == NilTimestamp
+func CompareBytes(b1 []byte, b2 []byte) int {
+	if len(b1) != len(b2) {
+		if len(b1) < len(b2) {
+			return -1
+		} else {
+			return 1
+		}
+	}
+
+	for i := range b1 {
+		if b1[i] != b2[i] {
+			if b1[i] < b2[i] {
+				return -1
+			} else {
+				return 1
+			}
+		}
+	}
+
+	return 0
 }
 
-func (n *NetworkInfo) Equal(other NetworkInfo) bool {
-	return n.LocalAddress == other.LocalAddress &&
-		n.RemoteAddress == other.RemoteAddress &&
-		n.Role == other.Role &&
-		n.SocketFamily == other.SocketFamily &&
-		n.IsActive() == other.IsActive()
+func EqualNetworkAddress(addr1 *sensorAPI.NetworkAddress, addr2 *sensorAPI.NetworkAddress) bool {
+	ad1 := adjustNetworkAddressForComparison(addr1)
+	ad2 := adjustNetworkAddressForComparison(addr2)
+
+	return ad1.EqualVT(ad2)
 }
 
-func (n *NetworkInfo) Less(other NetworkInfo) bool {
-	if n.LocalAddress != other.LocalAddress {
-		return n.LocalAddress < other.LocalAddress
+func LessNetworkAddress(addr1 *sensorAPI.NetworkAddress, addr2 *sensorAPI.NetworkAddress) bool {
+	comp := CompareBytes(addr1.GetAddressData(), addr2.GetAddressData())
+
+	if comp != 0 {
+		return comp < 0
 	}
 
-	if n.RemoteAddress != other.RemoteAddress {
-		return n.RemoteAddress < other.RemoteAddress
+	comp = CompareBytes(addr1.GetIpNetwork(), addr2.GetIpNetwork())
+
+	if comp != 0 {
+		return comp < 0
 	}
 
-	if n.Role != other.Role {
-		return n.Role < other.Role
+	return addr1.GetPort() < addr2.GetPort()
+}
+
+func LessNetworkConnection(conn1 *sensorAPI.NetworkConnection, conn2 *sensorAPI.NetworkConnection) bool {
```

Comment:
I could do something like convert the connections to a string using the generated `String` method and then compare them, but I think that might fail in some cases.